dev/com.ibm.ws.cdi.interfaces/src/io/openliberty/cdi/spi/WebSphereCDIExtensionMetaData.java

```diff
@@ -0,0 +1,74 @@
+/*******************************************************************************
+ * Copyright (c) 2015 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.cdi.spi;
+
+import java.util.Collections;
+import java.util.Set;
+
+import javax.enterprise.inject.spi.Extension;
+
+/**
+ * This is a *marker* interface for Weld Runtime extensions. Liberty features that wish to extend CDI will need to 
+ * need to register a service under this interface. 
+ *
+ * To use this class you must implement one of the two methods. If you implement getBeanClasses() all classes returned by that
+ * method will be registered with CDI and may be used normally by application code. If you implement getExtensions() then 
+ * all classes returned by that method will be treated as CDI extensions and run when creating the CDI container for each application.
+ * 
+ * Classes returned from getExtensions() must implement javax.enterprise.inject.spi.Extension. They do *not* and should not be listed in a 
+ * META-INF/services file. It is best practice to not put WebSphereCDIExtensionMetaData and Extension on the same class as that would result
```

Comment:
WebSphereCDIExtensionMetaData -> CDIExtensionMetaData. You should add the package for Extension. Also, you can say

Implementing CDIExtension or javax.enterprise.inject.spi.Extension is mutually exclusive. Otherwise, none desirable behaviour will occur.